tests/plugin-parser/test-mlops-plugin-parser.cc

```diff
@@ -0,0 +1,397 @@
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_res_version (handle, return_res_version);
+}
+
+/**
+ * @brief pluging-parser test fixture
+ */
+class PkgMgrInfoMockTestFixture : public ::testing::Test
+{
+  protected:
+  GModule *module;
+  GTestDBus *dbus;
+
+  /**
+   * @brief Set up the test fixture
+   */
+  void SetUp () override
+  {
+    g_autofree gchar *current_dir = g_get_current_dir ();
+    g_autofree gchar *services_dir
+        = g_build_filename (current_dir, "tests", "services", NULL);
+
+    dbus = g_test_dbus_new (G_TEST_DBUS_NONE);
+    ASSERT_TRUE (dbus != nullptr);
+
+    g_test_dbus_add_service_dir (dbus, services_dir);
+    g_test_dbus_up (dbus);
+
+    module = g_module_open ("plugin-parser/libmlops-plugin-parser.so", (GModuleFlags) G_MODULE_BIND_LAZY);
+    ASSERT_TRUE (module != nullptr);
+  }
+
+  /**
+   * @brief Tear down the test fixture.
+  */
+  void TearDown () override
+  {
+    g_module_close (module);
+    if (dbus) {
+      g_test_dbus_down (dbus);
+      g_object_unref (dbus);
+      dbus = nullptr;
+    }
+  }
+
+  public:
+  /**
+   * @brief Executing the function from the loaded module, plugin-parser library.
+   */
+  int exec_plugin_parser_func (const std::string func, const std::string pkgid, const std::string appid, GList *metadata) {
+    int (*func_ptr)(const char*, const char*, GList*);
+    if (!g_module_symbol (module, func.c_str (), (gpointer *) &func_ptr)) {
+      g_warning ("Failed to get symbol %s", func.c_str ());
+      return -1;
+    }
+
+    return func_ptr (pkgid.c_str (), appid.c_str (), metadata);
+  }
+};
+
+/**
+ * @brief Negative test case of pkgmgrinfo_pkginfo_get_pkginfo() failed.
+ */
+TEST_F (PkgMgrInfoMockTestFixture, test_n1)
+{
+  pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
+  std::string pkgid = "pkgid";
+  std::string appid = "appid";
+  GList *md_list = nullptr;
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_pkginfo (_, _))
+      .WillOnce (Return (PMINFO_R_ERROR));
+
+
+  EXPECT_EQ (exec_plugin_parser_func ("PKGMGR_MDPARSER_PLUGIN_INSTALL",pkgid, appid, md_list), -1);
+
+  delete pkgMgrInfoMockInstance;
+}
+
+/**
+ * @brief Negative test case of pkgmgrinfo_pkginfo_get_type() failed.
+ */
+TEST_F (PkgMgrInfoMockTestFixture, test_n2)
+{
+  pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
+  std::string pkgid = "pkgid";
+  std::string appid = "appid";
+  GList *md_list = nullptr;
+
+  EXPECT_CALL (*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_destroy_pkginfo (_))
+      .WillRepeatedly (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_pkginfo (_, _))
+      .WillOnce (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_type (_, _))
+      .WillOnce (Return (PMINFO_R_ERROR));
+
+  EXPECT_EQ (exec_plugin_parser_func ("PKGMGR_MDPARSER_PLUGIN_INSTALL",pkgid, appid, md_list), -1);
```

Comment:
Ditto.

How about compare `PMINFO_R_ERROR` instead of -1?

Reply:
Or, EXPECT_NE (exec_plugin_parser_func (), 0) :)

Reply:
I updated the code to use `EXPECT_NE`, PTAL